### PR TITLE
Add Bluemix deploy provider

### PR DESCRIPTION
**Pull request by adamkingit**
Subclass the cloudfoundry provider to allow users to select bluemix regions by name, without needing to know the api host. 

Add unit test coverage, as well as README.md documentation.

Example successful deployment at https://travis-ci.org/adamkingit/dpl/builds/184277695

### CODE REVIEW

File: dpl.gemspec [hunk 1]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'rspec', '~> 3.0.0'
   s.add_development_dependency 'rspec-its'
   s.add_development_dependency 'rake'
-  s.add_development_dependency 'json', '1.8.1'
+  s.add_development_dependency 'json', '1.8.2'
```
Comment (BanzaiMan) [inline dpl.gemspec]: JSON 1.8.1 is required for Ruby 1.9.3 support. See #513.

File: lib/dpl/provider/bluemix_cloudfoundry.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module DPL
+  class Provider
+    class BluemixCF < CloudFoundry
```
Comment (BanzaiMan) [inline lib/dpl/provider/bluemix_cloudfoundry.rb]: Is this really the right name to use?

**Comment by adamkingit** [comment]
Thanks for catching my mistake updating the json dependency. This is the first ruby I have done. I was having trouble with that package on Ubuntu 16, and I obviously chose the wrong solution.

File: lib/dpl/provider/bluemix_cloudfoundry.rb (hunk 2 above)
Comment (adamkingit) [inline lib/dpl/provider/bluemix_cloudfoundry.rb]: I am open to better/more consistent names with other dpl providers. I chose this name to leave open the opportunity for deploy providers to work with other Bluemix such as the container service. Would bluemix-cloudfoundry be more consistent?

**Comment by BanzaiMan** [comment]
Cool. Could you work on documentation? It should go to https://github.com/travis-ci/docs-travis-ci-com.

**Comment by adamkingit** [comment]
I renamed the class as suggested. I was, and am, getting errors on most environments when travis builds the pull request, but I am not sure how its related to my change. I pulled the latest travis-ci/dpl master onto my fork, but it make a difference. 

I also looked at the https://github.com/travis-ci/docs-travis-ci-com documentation, but don't see an obvious area to update. What do you think about a pull request to add a CONTRIBUTING.md to dpl, or did you have something else in mind?

**Comment by BanzaiMan** [comment]
Hi, @adamkingit, Sorry for the general lack of contribution guidance on docs. The deployment documentation should go to https://github.com/travis-ci/docs-travis-ci-com/tree/gh-pages/user/deployment.

**Comment by adamkingit** [comment]
I updated docs and cli for bluemix and submitted pull requests for:
https://github.com/travis-ci/travis.rb/pull/475
https://github.com/travis-ci/docs-travis-ci-com/pull/925

**Comment by adamkingit** [comment]
Documentation pull request https://github.com/travis-ci/docs-travis-ci-com/pull/925 now merged.

File: dpl.gemspec (hunk 1 above)
Comment (adamkingit) [inline dpl.gemspec]: Done

File: lib/dpl/provider/bluemix_cloudfoundry.rb (hunk 2 above)
Comment (adamkingit) [inline lib/dpl/provider/bluemix_cloudfoundry.rb]: Done

**Comment by BanzaiMan** [comment]
Will release it tomorrow.

**Pull request merged**